### PR TITLE
Update AbstractAdmin.php

### PR DESCRIPTION
**Pull request by axzx**
## Subject

If I have link with the id which is an array
```
list?_sort_order=desc&_sort_by=id&_page=1&_per_page=32&id[type]=&id[value]=&
```

I have problem when I use getSubject in getAccess.

### CODE REVIEW

File: Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -1669,7 +1669,9 @@ public function getSubject()
     {
         if ($this->subject === null && $this->request) {
             $id = $this->request->get($this->getIdParameter());
-            $this->subject = $this->getModelManager()->find($this->class, $id);
+            if (!\is_array($id)) {
```
Comment (greg0ire) [inline Admin/AbstractAdmin.php]: You don't need this backslash, do you? There is no `is_array` method in the current namespace AFAIK

**Comment by greg0ire** [comment]
> list?_sort_order=desc&_sort_by=id&_page=1&_per_page=32&id[type]=&id[value]=&

I'm confused… how can you even have a link like that?

**Comment by axzx** [comment]
Sorry, 
my mistake.
After batch action I have 
```
return $this->redirect($this->admin->generateUrl('list', $this->admin->getFilterParameters()));
```

**Comment by greg0ire** [comment]
Oh I see now… it refers to the type of filtering… something is obviously wrong there, I don't think filtering by id makes sense. Why do you have a filter on id?

**Comment by axzx** [comment]
why not?:)
Sometimes an administrator wants to search for a specific record knowing its id.

**Comment by greg0ire** [comment]
Id should be something internal, but let's say it's ok for the sake of the argument. Shouldn't the url look like `filter[id][type]` rather than just `id[type]` ?

**Comment by axzx** [comment]
You are absolutely right!

Should be:
```
return $this->redirect($this->admin->generateUrl('list', array('filter' => $this->admin->getFilterParameters())));
```


Look here:
http://symfony.com/doc/current/bundles/SonataAdminBundle/reference/batch_actions.html


is there an example:
```
$this->admin->generateUrl('list', $this->admin->getFilterParameters())
```

**Comment by greg0ire** [comment]
OMG the doc is wrong then! Can you please try that, see if it works, and make a PR?

**Pull request closed without merging**